compiler/src/iree/compiler/Dialect/VM/Conversion/VMToEmitC/ConvertVMToEmitC.cpp

```diff
@@ -1243,22 +1243,33 @@ void updateResultUses(Operation *op, ConversionPatternRewriter &rewriter,
   }
 }
 
+template <typename OpTy>
+class EmitCConversionPattern : public OpConversionPattern<OpTy> {
+public:
+  using Adaptor = typename OpTy::Adaptor;
+  using OpConversionPattern<OpTy>::OpConversionPattern;
+
+  EmitCConversionPattern(const TypeConverter &typeConverter,
+                         MLIRContext *context)
+      : OpConversionPattern<OpTy>(typeConverter, context) {}
+};
```

Comment:
Something in here is ambiguous, breaking builds on Windows / MSVC

https://github.com/openxla/iree/actions/runs/6614754015/job/17965409798#step:8:7604

```
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\memory(3393): error C2668: 'mlir::iree_compiler::`anonymous-namespace'::BranchOpConversion::BranchOpConversion': ambiguous call to overloaded function
C:\a\iree\iree\compiler\src\iree\compiler\Dialect\VM\Conversion\VMToEmitC\ConvertVMToEmitC.cpp(2878): note: could be 'mlir::iree_compiler::`anonymous-namespace'::BranchOpConversion::BranchOpConversion(const mlir::TypeConverter &,mlir::MLIRContext *)', which inherits 'mlir::iree_compiler::`anonymous-namespace'::EmitCConversionPattern<mlir::iree_compiler::IREE::VM::BranchOp>::EmitCConversionPattern(const mlir::TypeConverter &,mlir::MLIRContext *)' via base class 'mlir::iree_compiler::`anonymous-namespace'::EmitCConversionPattern<mlir::iree_compiler::IREE::VM::BranchOp>'
C:\a\iree\iree\compiler\src\iree\compiler\Dialect\VM\Conversion\VMToEmitC\ConvertVMToEmitC.cpp(2878): note: or       'mlir::iree_compiler::`anonymous-namespace'::BranchOpConversion::BranchOpConversion(const mlir::TypeConverter &,mlir::MLIRContext *,mlir::PatternBenefit)', which inherits 'mlir::OpConversionPattern<OpTy>::OpConversionPattern(const mlir::TypeConverter &,mlir::MLIRContext *,mlir::PatternBenefit)' via base class 'mlir::iree_compiler::`anonymous-namespace'::EmitCConversionPattern<mlir::iree_compiler::IREE::VM::BranchOp>'
        with
        [
            OpTy=mlir::iree_compiler::IREE::VM::BranchOp
        ]
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.35.32215\include\memory(3393): note: while trying to match the argument list '(mlir::iree_compiler::IREE::VM::EmitCTypeConverter, mlir::MLIRContext *)'
C:\a\iree\iree\third_party\llvm-project\mlir\include\mlir/IR/PatternMatch.h(275): note: see reference to function template instantiation 'std::unique_ptr<T,std::default_delete<T>> std::make_unique<T,mlir::iree_compiler::IREE::VM::EmitCTypeConverter&,mlir::MLIRContext*&,0>(mlir::iree_compiler::IREE::VM::EmitCTypeConverter &,mlir::MLIRContext *&)' being compiled
        with
        [
            T=mlir::iree_compiler::`anonymous-namespace'::BranchOpConversion
        ]
C:\a\iree\iree\third_party\llvm-project\mlir\include\mlir/IR/PatternMatch.h(1850): note: see reference to function template instantiation 'std::unique_ptr<T,std::default_delete<T>> mlir::RewritePattern::create<T,mlir::iree_compiler::IREE::VM::EmitCTypeConverter&,mlir::MLIRContext*&>(mlir::iree_compiler::IREE::VM::EmitCTypeConverter &,mlir::MLIRContext *&)' being compiled
        with
        [
            T=mlir::iree_compiler::`anonymous-namespace'::BranchOpConversion
        ]
```

Revert then fix-forward? Or is there a simple fix?

Reply:
Sent https://github.com/openxla/iree/pull/15267